src/azul/bigquery_reservation.py

```diff
@@ -96,6 +117,26 @@ def is_active(self) -> Optional[bool]:
         else:
             return None
 
+    @property
+    def age(self) -> Optional[timedelta]:
```

Comment:
Age depends on the current time. In the functional paradigm this would be considered a non-strict function. When it's straight-forward prefer writing functions that are strict in the functional sense as it simplifies reasoning about them and enables caching. Also don't care for use of timedelta. We typically use Unix time (fractional seconds and calculate seconds. `timedelta(minutes=30)` is longer than `30 * 60` and leap seconds can be neglected for this.